### ISSUE TITLE
Improve wording and flow of restart EH blocker

### ISSUE THREAD

**Issue by jrieken** [microsoft/vscode#233912]
* run a debug session
* update an extension and trigger EH restart
* 😕 be confused about the wording here 

![Image](https://github.com/user-attachments/assets/0227e649-2c8c-4d99-96af-4b690fc10096)

How about a less technical language. Like 

```
Restart Extension Blocked

A debug session is still running and would be terminated otherwise

[OK]
[Restart Anyways]

```

**Comment by bpasero** [comment]
@sandy081 the label "feature request" serves a purpose: having it means, this issue is getting verified during endgame. As far as I know, "polish" does not mean the issue is picked up by our endgame queries.

**Comment by bpasero** [comment]
I went over all the messages and rewrote them to this:

**Reasons to restart:**
- Workspace trust change
- Extension enablement change
- Workspace folder change
- Explicit request
- Switching to a profile
- Opening a multi-root workspace

**Reasons to block:**
- An extension provided editor for '{0}' is still open.
- An extension provided text editor for '{0}' is still open.
- A debug session is still running.

As such the dialog becomes this:
- title: `Extension restart was prevented but is required for: {reason to restart}. Do you want to proceed anyways?`
- detail: `Resaon(s): {reason to block}`

**Comment by sandy081** [comment]
> As far as I know, "polish" does not mean the issue is picked up by our endgame queries.

I think this is polishing the text and `polish` label makes more sense than feature request. I would question why polish items are not getting verified, they are valid issue type.

**Comment by bpasero** [comment]
@sandy081 yes, proposed this in https://github.com/microsoft/vscode-internalbacklog/issues/5281

**Comment by jrieken** [comment]
Reopening because I think the wording still isn't great

* the title (bold part) is super long, the whole "... was prevented but is required: <some action>" makes it hard to parse
* the description has again a form-like attitude with "Reason: <some reasons>


<img width="571" alt="image" src="https://github.com/user-attachments/assets/5e488df9-06e6-4552-b50c-a0b02b2985ac" />


Why do we need to spell out the action at all? This should be the thing that I just did anyways, right? Can't we have


```
Restarting Extensions

A running debug session would be terminated.

[Abort]
[Proceed Anyways]
```
